lib/commons/matches/from-definition.js

```diff
@@ -0,0 +1,48 @@
+/* global matches */
+const matchers = [
+  'nodeName',
+  'attributes',
+  'properties',
+  'condition'
+]
+
+/**
+ * Check if a node matches some definition
+ * 
+ * Example:
+ * ```js
+ * matches.fromDefinition(node, {
+ *   nodeName: ['div', 'span']
+ *   attributes: {
+ *     'aria-live': 'assertive'
+ *   }
+ * })
+ * ```
+ * 
+ * @param {HTMLElement|VirtualNode} node
+ * @param {Object|Array<Object>} definition
+ * @returns {Boolean}
+ */
+matches.fromDefinition = function matchFromDefinition (node, definition) {
+  node = node.actualNode || node;
+	if (Array.isArray(definition)) {
+		return definition.some(definition => matches(node, definition));
+  }
+	if (typeof definition === 'string') {
+		return axe.utils.matchesSelector(node, definition);
+  }
+
+  return Object.keys(definition)
+  .every(matcherName => {
+    if (!matchers.includes(matcherName)) {
+      throw new Error(`Unknown matcher type "${matcherName}"`)
+    }
+    const matchMethod = matches[matcherName]
```

Comment:
Can you add a comment here about what is happening with `matches`? It's so overloaded it's hard to tell exactly.

Reply:
Yeah, that was the challenge writing this code. I'll add in comments.